gplugins/gmsh/xy_xsection_mesh.py

```diff
@@ -106,9 +105,8 @@ def xy_xsection_mesh(
         layer_port_delimiter: Delimiter to use for new layers generated for ports: "layer{delimiter}port_name".
     """
     if port_names:
-        mesh_component = gf.Component()
-        _ = mesh_component << union(component, by_layer=True)
-        mesh_component.add_ports(component.get_ports_list())
+        mesh_component = component
+        mesh_component.add_ports(component.ports)
```

Comment:
I think mesh_component should be a new component, to which we add the polygons and ports of component

right now we get the following error on the test on these lines:

`ValueError: add_port() Port name 'o1' exists in 'Unnamed_57c8ad0b'`

Reply:
i see, yes, we can make a new component

Reply:
does the new changes work?

Reply:
is it running using gdsfactory8? We're getting `'Component' object has no attribute 'dbbox'` everywhere

Reply:
i think i fixed this one